subliminal/providers/itasa.py

```diff
@@ -0,0 +1,390 @@
+# -*- coding: utf-8 -*-
+import copy
+
+import io
+
+import logging
+
+from babelfish import Language
+
+from requests import Session
+
+from zipfile import ZipFile, is_zipfile
+
+from . import ParserBeautifulSoup, Provider, get_version
+
+from .. import __version__
+
+from .. cache import SHOW_EXPIRATION_TIME, region
+
+from .. exceptions import AuthenticationError, ConfigurationError, TooManyRequests
+
+from .. subtitle import (Subtitle, fix_line_ending, guess_matches, guess_properties, sanitize_string,
+                         sanitized_string_equal)
+
+from .. video import Episode
+
+logger = logging.getLogger(__name__)
+
+
+class ItaSASubtitle(Subtitle):
+    provider_name = 'itasa'
+
+    def __init__(self, sub_id, series, season, episode, format, full_data):
+        super(ItaSASubtitle, self).__init__(Language('ita'))
+        self.sub_id = sub_id
+        self.series = series
+        self.season = season
+        self.episode = episode
+        self.format = format
+        self.full_data = full_data
+
+    @property
+    def id(self):
+        return self.sub_id
+
+    @property
+    def format(self):
+        return self._format
+
+    @format.setter
+    def format(self, value):
+        if not value:
+            self._format = 'HDTV'
+        elif value and value.lower() == 'normale':
+            self._format = 'HDTV'
+        else:
+            self._format = value
+
+    def get_matches(self, video, hearing_impaired=False):
+        matches = super(ItaSASubtitle, self).get_matches(video, hearing_impaired=hearing_impaired)
+
+        # series
+        if video.series and sanitized_string_equal(self.series, video.series):
+            matches.add('series')
+        # season
+        if video.season and self.season == video.season:
+            matches.add('season')
+        # episode
+        if video.episode and self.episode == video.episode:
+            matches.add('episode')
+        # format
+        if video.format and video.format.lower() in self.format.lower():
+            matches.add('format')
+        if not video.format and video.format.lower() in 'hdtv':
+            matches.add('format')
+        # other properties
+        matches |= guess_matches(video, guess_properties(self.full_data), partial=True)
+
+        return matches
+
+
+class ItaSAProvider(Provider):
+    languages = {Language('ita')}
+
+    video_types = (Episode,)
+
+    server_url = 'https://api.italiansubs.net/api/rest/'
+
+    apikey = 'd86ad6ec041b334fac1e512174ee04d5'
+
+    itasa_formats = {'1080i': '1080i',
+                     '1080p': '1080p',
+                     '720p': '720p',
+                     'bdrip': 'bdrip',
+                     'bluray': 'bluray',
+                     'dvdrip': 'dvdrip',
+                     'hdtv': 'normale',
+                     'hr': 'hr',
+                     'web-dl': 'web-dl'}
```

Comment:
I want to make a generic mechanism, but maybe is better to convert on the fly in the code the create the subtitle instance